src/form/acp.jl

```diff
@@ -149,7 +149,53 @@ function constraint_ohms_yt_to{T <: AbstractACPForm}(pm::GenericPowerModel{T}, f
     t_to = pm.var[:t][t_bus]
 
     c1 = @NLconstraint(pm.model, p_to == g*v_to^2 + (-g*tr-b*ti)/tm*(v_to*v_fr*cos(t_to-t_fr)) + (-b*tr+g*ti)/tm*(v_to*v_fr*sin(t_to-t_fr)) )
-    c2 = @NLconstraint(pm.model, q_to == -(b+c/2)*v_to^2 - (-b*tr+g*ti)/tm*(v_to*v_fr*cos(t_fr-t_to)) + (-g*tr-b*ti)/tm*(v_to*v_fr*sin(t_to-t_fr)) )
+    c2 = @NLconstraint(pm.model, q_to == -(b+c/2)*v_to^2 - (-b*tr+g*ti)/tm*(v_to*v_fr*cos(t_to-t_fr)) + (-g*tr-b*ti)/tm*(v_to*v_fr*sin(t_to-t_fr)) )
+    return Set([c1, c2])
+end
+
+"""
+Creates Ohms constraints for shiftable PSTs (yt post fix indicates that Y and T values are in rectangular form)
+
+```
+p[f_idx] == g*v_fr^2 + (-g)*(v_fr*v_to*cos((t_fr - t_shift_fr) - (t_to - t_shift_to))) + (-b)*(v_fr*v_to*sin((t_fr - t_shift_fr) - (t_to - t_shift_to)))
+q[f_idx] == -(b+c/2)*v_fr^2 - (-b)*(v_fr*v_to*cos((t_fr - t_shift_fr) - (t_to - t_shift_to))) + (-g)*(v_fr*v_to*sin((t_fr - t_shift_fr) - (t_to - t_shift_to)))
+```
+"""
+function constraint_ohms_yt_from_shiftable{T <: AbstractACPForm}(pm::GenericPowerModel{T}, f_bus, t_bus, f_idx, t_idx, g, b, c)
+    p_fr = pm.var[:p][f_idx]
+    q_fr = pm.var[:q][f_idx]
+    v_fr = pm.var[:v][f_bus]
+    v_to = pm.var[:v][t_bus]
+    t_fr = pm.var[:t][f_bus]
+    t_to = pm.var[:t][t_bus]
+    t_shift_fr = pm.var[:t_shift][f_idx]
+    t_shift_to = pm.var[:t_shift][t_idx]
+
+    c1 = @NLconstraint(pm.model, p_fr == g*v_fr^2 + (-g)*(v_fr*v_to*cos((t_fr - t_shift_fr) - (t_to - t_shift_to))) + (-b)*(v_fr*v_to*sin((t_fr - t_shift_fr) - (t_to - t_shift_to))))
```

Comment:
It matters on which side you physically put the voltage magnitude tap changer (easy to see that losses are impacted). You see both options being modeled also [elsewhere](https://nl.mathworks.com/help/physmod/sps/examples/oltc-regulating-transformer-phasor-model.html). Typically, an OLTC will be on the 'from' side - the OLTC device will have to handle much less current. We could implement a 'from'-only formulation here and add the extensions to PowerModelsAnnex.jl. 

For a PST it's a matter of signs (shift(to-> from) = -shift(from->to)). But it feels weird having asymmetric PST parameters if the OLTC part is symmetric. 

In any case, I'm inclined to rename the default fields `tap` to `tap_fr` and `shift` to `shift_fr`, to be more clear w.r.t. what the matpower parameters actually _represent_ (cfr. [Fig 3-1](http://www.pserc.cornell.edu/matpower/MATPOWER-manual.pdf)). 

Furthermore, we aim to add active power losses (no-load) in the transformer model as well (e.g. as shunt conductance of the transformer), to satisfy modelling requirements for distribution systems (such losses are not equally negligible in distribution as in transmission). Which direction would you want to go here:

- add conductance to line equations in general (g_shunt = 0 for backwards compatibility)
- separate transformer models from normal line models with support for shunt conductance

I feel that extensions should include the matpower line model at least as an edge case (for backwards compatibility). But sticking to a simple pi model implies certain limitations (i.e. equally focussed on transmission system modeling accuracy tradeoffs). What's your take on this?